features/net/network-socket/NetworkInterface.h

```diff
@@ -36,6 +38,42 @@ class NetworkInterface {
      */
     virtual const char *get_ip_address() = 0;
 
+    /** Translates a hostname to an IP address
+     *
+     *  The hostname may be either a domain name or an IP address. If the
+     *  hostname is an IP address, no network transactions will be performed.
+     *  
+     *  If no stack-specific DNS resolution is provided, the hostname
+     *  will be resolve using a UDP socket on the stack. 
+     *
+     *  @param address  Destination for the host SocketAddress
+     *  @param host     Hostname to resolve
+     *  @return         0 on success, negative error code on failure
+     */
+    virtual int gethostbyname(const char *host, SocketAddress *address);
```

Comment:
Is there any reason to make this virtual? In what case would gethostbyname be overridden by an interface?